flink-libraries/flink-table/src/test/scala/org/apache/flink/table/api/scala/stream/sql/SqlITCase.scala

```diff
@@ -411,6 +411,128 @@ class SqlITCase extends StreamingWithStateTestBase {
     assertEquals(expected.sorted, StreamITCase.testResults.sorted)
   }
 
+  @Test
+  def testBoundPartitionedEventTimeWindowWithRange(): Unit = {
+    val data = Seq(
+      Left((1000L, (1L, 1, "Hello"))),
+      Left((1000L, (1L, 1, "Hello"))),
```

Comment:
The problem of copying the forwarded fields is not captured by the tests, because all rows for the same timestamp are identical.